forge/test/operators/utils/compat.py

```diff
@@ -22,12 +24,198 @@ class TestDevice:
+        torch.int32: (-(2**31), 2**31 - 1),
+    }
+
+    @classmethod
+    def get_value_range(
+        cls,
+        dev_data_format: forge.DataFormat,
+        dtype: torch.dtype,
+        value_range: Optional[OperatorParameterTypes.RangeValue] = None,
+    ) -> OperatorParameterTypes.RangeValue:
+        if dev_data_format in cls.data_format_ranges:
+            data_format_ranges = cls.data_format_ranges[dev_data_format]
+        else:
+            raise ValueError(f"Unsupported range for dev data format: {dev_data_format}")
+        if dtype in cls.dtype_ranges:
+            dtype_ranges = cls.dtype_ranges[dtype]
+        else:
+            raise ValueError(f"Unsupported range for dtype: {dtype}")
+        range = cls.intersect_ranges(data_format_ranges, dtype_ranges)
+        if value_range is not None:
+            range = cls.intersect_ranges(range, value_range)
+        return range
+
+    @classmethod
+    def intersect_ranges(
+        cls, range1: OperatorParameterTypes.RangeValue, range2: OperatorParameterTypes.RangeValue
+    ) -> OperatorParameterTypes.RangeValue:
+
+        low = None
+        if range1[0] is None and range2[0] is None:
+            low = None
+        else:
+            if range1[0] is None:
+                low = range2[0]
+            elif range2[0] is None:
+                low = range1[0]
+            else:
+                low = max(range1[0], range2[0])
+
+        high = None
+        if range1[1] is None and range2[1] is None:
+            high = None
+        else:
+            if range1[1] is None:
+                high = range2[1]
+            elif range2[1] is None:
+                high = range1[1]
+            else:
+                high = min(range1[1], range2[1])
+
+        return low, high
+
+    @classmethod
+    def get_dtype_for_df(cls, dev_data_format: forge.DataFormat = None) -> torch.dtype:
+
+        dtype: torch.dtype
+
+        if dev_data_format is None:
+            dtype = None
+        else:
+            # dtype = torch.float32
+            if dev_data_format in cls.dev_data_format_to_dtype:
+                dtype = cls.dev_data_format_to_dtype[dev_data_format]
+            else:
+                raise ValueError(f"Unsupported dtype for dev data format: {dev_data_format}")
+
+        return dtype
+
+    @classmethod
+    def move_from_small_to_big_value_range(cls, input: torch.Tensor, min, max) -> torch.Tensor:
+        return min + (max - min) * input
+
+    @classmethod
+    def get_random_sign(cls, generator: torch.Generator) -> int:
+        return 2 * torch.randint(0, 2, (1,), generator=generator).item() - 1
+
+    @classmethod
+    def get_random_torch_input(
+        cls,
+        dev_data_format: forge.DataFormat,
+        dtype: torch.dtype,
+        input_shape: TensorShape,
+        generator: torch.Generator,
+        value_range: Optional[OperatorParameterTypes.RangeValue] = None,
+    ) -> List[torch.Tensor]:
+
+        if dtype is None:
```

Comment:
this if elseif will tend to explode with new data formats. Can we make some helper functions for generating random inputs for each category?

Reply:
Dtypes are grouped as floats, integers and bools so it's not expected to have more ifs with new dataformats. However since too many ifs in single function is already unnecessary hard to read there is new method `get_signed_value_range`

Reply:
Actually this whole method `get_signed_value_range` looks not needed. I will remove it

Reply:
Removed